app/controllers/project_roles_controller.rb

```diff
@@ -39,4 +41,8 @@ def create_params
   def update_params
     params.require(:project_role).permit(:role_id)
   end
+
+  def reset_access_request_flag(user_id)
+    User.find(user_id).update!(access_request_pending: false)
```

Comment:
I think you might be able to do pass `new_role.user` as argument and then do:
```
user.update!(...)
```

Sorry, I mentioned the user_id before but I just noticed that you should be able to get the User from the UserProjectRole.